tools/testing/e2e/iree-e2e-matmul-test.cc

```diff
@@ -21,184 +19,12 @@
-  int check_every = 1;
-  if (FLAG_max_elements_to_check) {
-    check_every = ((m_size * n_size) + FLAG_max_elements_to_check - 1) /
-                  FLAG_max_elements_to_check;
-    if (check_every < 1) check_every = 1;
-    if (check_every > 1)
-      while ((n_size % check_every) == 0) ++check_every;
-  }
-  return check_every;
-}
-
-// Defines the type of a primitive value.
-typedef enum iree_e2e_test_value_type_e {
-  // Not a value type.
-  IREE_E2E_TEST_VALUE_TYPE_NONE = 0,
-  // int8_t.
-  IREE_E2E_TEST_VALUE_TYPE_I8 = 1,
-  // int16_t.
-  IREE_E2E_TEST_VALUE_TYPE_I16 = 2,
-  // int32_t.
-  IREE_E2E_TEST_VALUE_TYPE_I32 = 3,
-  // int64_t.
-  IREE_E2E_TEST_VALUE_TYPE_I64 = 4,
-  // halft_t.
-  IREE_E2E_TEST_VALUE_TYPE_F16 = 5,
-  // float.
-  IREE_E2E_TEST_VALUE_TYPE_F32 = 6,
-  // double.
-  IREE_E2E_TEST_VALUE_TYPE_F64 = 7,
-  // bfloat16
-  IREE_E2E_TEST_VALUE_TYPE_BF16 = 8,
-} iree_e2e_test_value_type_t;
-
-// Maximum size, in bytes, of any value type we can represent.
-#define IREE_E2E_TEST_VALUE_STORAGE_SIZE 8
-
-// A variant value type.
-typedef struct iree_e2e_test_value_t {
-  iree_e2e_test_value_type_t type;
-  union {
-    int8_t i8;
-    int16_t i16;
-    int32_t i32;
-    int64_t i64;
-    float f32;
-    uint16_t f16_u16;
-    uint16_t bf16_u16;
-    double f64;
-    uint8_t value_storage[IREE_E2E_TEST_VALUE_STORAGE_SIZE];  // max size of all
-                                                              // value types
-  };
-} iree_e2e_test_value_t;
-
-static inline iree_e2e_test_value_t iree_e2e_test_value_make_none() {
-  iree_e2e_test_value_t result;
-  result.type = IREE_E2E_TEST_VALUE_TYPE_NONE;
-  return result;
-}
-
-static inline iree_e2e_test_value_t iree_e2e_test_value_make_i8(int8_t value) {
-  iree_e2e_test_value_t result;
-  result.type = IREE_E2E_TEST_VALUE_TYPE_I8;
-  result.i8 = value;
-  return result;
-}
-
-static inline iree_e2e_test_value_t iree_e2e_test_value_make_i16(
-    int16_t value) {
-  iree_e2e_test_value_t result;
-  result.type = IREE_E2E_TEST_VALUE_TYPE_I16;
-  result.i16 = value;
-  return result;
-}
-
-static inline iree_e2e_test_value_t iree_e2e_test_value_make_i32(
-    int32_t value) {
-  iree_e2e_test_value_t result;
-  result.type = IREE_E2E_TEST_VALUE_TYPE_I32;
-  result.i32 = value;
-  return result;
-}
-
-static inline iree_e2e_test_value_t iree_e2e_test_value_make_f16(
-    uint16_t value) {
-  iree_e2e_test_value_t result;
-  result.type = IREE_E2E_TEST_VALUE_TYPE_F16;
-  result.f16_u16 = value;
-  return result;
-}
-
-static inline iree_e2e_test_value_t iree_e2e_test_value_make_bf16(
-    uint16_t value) {
-  iree_e2e_test_value_t result;
-  result.type = IREE_E2E_TEST_VALUE_TYPE_BF16;
-  result.bf16_u16 = value;
-  return result;
-}
-
-static inline iree_e2e_test_value_t iree_e2e_test_value_make_f32(float value) {
-  iree_e2e_test_value_t result;
-  result.type = IREE_E2E_TEST_VALUE_TYPE_F32;
-  result.f32 = value;
-  return result;
-}
+#include "test_utils.h"
```

Comment:
nit: We typically fully qualify include paths, but that might not be set up for the `tools/` top level directory. I think this should be something like

```suggestion
#include "tools/testing/test_utils.h"
```

If that doesn't work, then it's fine to keep it as-is (at least until there are multiple header files)

https://google.github.io/styleguide/cppguide.html#Names_and_Order_of_Includes
> All of a project's header files should be listed as descendants of the project's source directory without use of UNIX directory aliases . (the current directory) or .. (the parent directory)

Reply:
It doesn't work as of now. I can consider this for a future update.

Reply:
Ah, maybe `tools/testing/e2e/test_utils.h`

Reply:
It works. I should have incorporated that. Apologies!